core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/ApacheSender43.java

```diff
@@ -71,15 +84,29 @@ public void dispose(HttpResponse response) {
     @Override
     public void close() {
         try {
-            httpClient.close();
+            ((CloseableHttpClient)getHttpClient()).close();
```

Comment:
Why not change the interface ApacheSender itself and modify to get the CloseableHttpClient. I am more inclined towards deprecating the support for ApacheHttpClient42 also which I believe is very old and even our SDK doesn't use it anywhere. I would like to avoid this cast.

Reply:
Resolved